core/domain/story_services.py

```diff
@@ -967,3 +968,67 @@ def record_completed_node_in_story_context(
         progress_model.completed_node_ids.append(node_id)
         progress_model.update_timestamps()
         progress_model.put()
+
+
+def get_chapter_notification_dicts() -> Dict[str, List[Dict[
+    str, Sequence[str]]]]:
+    """Returns a dict of behind schedule and upcoming chapters."""
```

Comment:
Missing args/returns/etc. -- and please use domain objects instead.

Reply:
Done